spec/controllers/admin/football_pitches_controller_spec.rb

```diff
@@ -0,0 +1,184 @@
+require 'rails_helper'
+include SessionsHelper
+
+RSpec.describe Admin::FootballPitchesController, type: :controller do
+  let(:user) { create :user }
```

Comment:
phân biệt let vs let!

Reply:
Sự khác biệt giữa let và let! nằm ở thời điểm thực thi. let được thực hiện khi được gọi trong khối `it` còn let! thực sự không cần gọi nó vẫn chạy trước khi khối `it` được thực thi, hay có cách nói khác là chỉ cần spec block được chạy thì let! đã được thực thi rồi.

Reply:
khi nào gọi thì khối lệnh trong let ms dc thực thi để tạo dữ liệu : lazy loading -> tránh tạo thừa dữ liệu k dùng đến
còn let! là cứ định nghĩa chạy đến là nó dc tạo bất kể có dc gọi hay k?